src/components/Coldcard/ColdcardExtendedPublicKeyImporter.jsx

```diff
@@ -1,101 +1,83 @@
-    };
-  }
-
-  componentDidMount = () => {
-    const { extendedPublicKeyImporter, validateAndSetBIP32Path } = this.props;
-    const { COLDCARD_MULTISIG_BIP32_PATH } = this.state;
-    if (extendedPublicKeyImporter.method === COLDCARD) {
-      validateAndSetBIP32Path(
-        COLDCARD_MULTISIG_BIP32_PATH,
-        () => {},
-        () => {},
-        {}
-      );
-    }
-  };
-
-  componentDidUpdate(prevProps) {
-    const { validateAndSetBIP32Path, network, addressType } = this.props;
-    const coldcardBIP32Path = this.getColdcardBip32Path();
-
-    // Any updates to the network/addressType we should set the BIP32Path
-    if (
-      prevProps.network !== network ||
-      prevProps.addressType !== addressType
-    ) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ COLDCARD_MULTISIG_BIP32_PATH: coldcardBIP32Path });
-      validateAndSetBIP32Path(
-        coldcardBIP32Path,
-        () => {},
-        () => {}
-      );
-    }
-  }
-
+const ColdcardExtendedPublicKeyImporter = ({
+  extendedPublicKeyImporter,
+  validateAndSetExtendedPublicKey,
+  validateAndSetBIP32Path,
+  validateAndSetRootFingerprint,
+  addressType,
+  network,
+  defaultBIP32Path,
+}) => {
   // Unfortunately not possible to use our Multisig P2SH ROOT on a Coldcard atm
   // because they do not allow us to export m/45'/{0-1}'/0' yet.
-  getColdcardBip32Path = () => {
-    const { network, addressType, defaultBIP32Path } = this.props;
+  const getColdcardBip32Path = () => {
     const coinPath = network === MAINNET ? "0" : "1";
     const coldcardP2SHPath = `m/45'/${coinPath}/0`;
     return addressType === P2SH ? coldcardP2SHPath : defaultBIP32Path;
   };
 
-  render = () => {
-    const {
-      extendedPublicKeyImporter,
-      validateAndSetExtendedPublicKey,
-      validateAndSetBIP32Path,
-      validateAndSetRootFingerprint,
-      addressType,
-      network,
-    } = this.props;
-    const { extendedPublicKeyError, COLDCARD_MULTISIG_BIP32_PATH } = this.state;
-    return (
-      <Box mt={2}>
-        <FormGroup>
-          <IndirectExtendedPublicKeyImporter
-            extendedPublicKeyImporter={extendedPublicKeyImporter}
-            validateAndSetExtendedPublicKey={validateAndSetExtendedPublicKey}
-            validateAndSetBIP32Path={validateAndSetBIP32Path}
-            validateAndSetRootFingerprint={validateAndSetRootFingerprint}
-            addressType={addressType}
-            network={network}
-            resetBIP32Path={this.resetColdcardBIP32Path}
-            defaultBIP32Path={COLDCARD_MULTISIG_BIP32_PATH}
-            Reader={ColdcardJSONReader}
-          />
-          <FormHelperText error>{extendedPublicKeyError}</FormHelperText>
-        </FormGroup>
-      </Box>
+  const [coldcardMultisigBIP32Path, setColdcardMultisigBIP32Path] = useState(
+    getColdcardBip32Path()
+  );
+  const [extendedPublicKeyError, setExtendedPublicKeyError] = useState("");
```

Comment:
`extendedPublicKeyError` is initialized as `""`, but its value is never updated. It is only used once in this file on line 76 ( `<FormHelperText error>{extendedPublicKeyError}</FormHelperText>`). I didn't want to change any functionality, but this might be a bug worth looking into.

Reply:
Nice. good catch.

Reply:
Would it be appropriate to remove the render call for <FormHelperText /> on line 76 then? Or would you prefer to leave it in and plan to update the state as a TODO?

Reply:
Yeah, I think that's reasonable.

Reply:
Which one is reasonable?

Reply:
Oh ha, sorry about that. Multitasking. I'm happy to remove it. I agree with your evaluation that it doesn't seem to be doing anything.

Reply:
Updated in [4c7681f](https://github.com/unchained-capital/caravan/pull/342/commits/4c7681fe2badd60264d2566255b8660f7e6c4151)

Reply:
No problem! Just wanted to make sure I was doing what you wanted.